test/integration/enumerate_databases.test.ts

```diff
@@ -141,6 +142,64 @@ describe('listDatabases()', function () {
     );
   });
 
+  describe('nameOnly option', function () {
+    let client: MongoClient;
+    const DBS = 10;
+    const nameOnlyOptions = [true, false, undefined];
+
+    beforeEach(async function () {
+      client = await this.configuration.newClient({}, { monitorCommands: true }).connect();
+      for (let i = 0; i < DBS; i++) {
+        const db = client.db(`testDb_${i}`);
+        await db.collection('test').insertOne({ a: 1 });
+      }
+    });
+
+    afterEach(async function () {
+      if (client) {
+        for (let i = 0; i < DBS; i++) {
+          await client.db(`testDb_${i}`).dropDatabase();
+        }
+
+        await client.close();
+      }
+    });
+
+    for (const nameOnly of nameOnlyOptions) {
+      context(`when options.nameOnly is ${nameOnly ?? 'not defined'}`, function () {
+        it(`sends command ${optionToExpecation(nameOnly)}`, async function () {
+          const promise = once(client, 'commandStarted');
+          await client.db().admin().listDatabases({ nameOnly });
+
+          const commandStarted = (await promise)[0];
+          expect(commandStarted.command).to.haveOwnProperty('listDatabases', 1);
+
+          switch (nameOnly) {
+            case true:
+              expect(commandStarted.command).to.have.property('nameOnly', true);
+              break;
+            case false:
+              expect(commandStarted.command).to.have.property('nameOnly', false);
+              break;
+            case undefined:
+              expect(commandStarted.command).to.not.have.property('nameOnly');
+              break;
+          }
+        });
+      });
+    }
+
+    function optionToExpecation(nameOnly?: boolean): string {
+      switch (nameOnly) {
+        case true:
+          return 'with nameOnly = true';
+        case false:
+          return 'with nameOnly = false';
+        case undefined:
+          return 'without nameOnly field';
+      }
+    }
```

Comment:
optional - functions like this that just map one value to another I think are better expressed as maps.  they're effectively the same, so feel free to ignore.

```suggestion
    const optionToExpecation = { 
      true: 'with nameOnly = true',
      false: 'with nameOnly = false',
      undefined: 'without nameOnly field'
   }
   
   optionToExpectation(String(<option>))
```